llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -1723,7 +1739,21 @@ void LoopVectorizationPlanner::updateLoopMetadataAndProfileInfo(
   // For scalable vectorization we can't know at compile time how many
   // iterations of the loop are handled in one vector iteration, so instead
   // use the value of vscale used for tuning.
-  setProfileInfoAfterUnrolling(OrigLoop, VectorLoop, OrigLoop, EstimatedVFxUF);
+  if (OrigAverageTripCount) {
```

Comment:
Early exit, as originally in llvm::setProfileInfoAfterUnrolling()?

Reply:
Done thanks!